src/javascript/_common/base/livechat.js

```diff
@@ -46,10 +46,11 @@ const LiveChat = (() => {
 
         setSessionVariables();
         setNameEmail();
-
-        window.LC_API.on_chat_ended = () => {
-            setNameEmail();
-        };
+        if (window.LC_API && window.LC_API.on_chat_ended) {
```

Comment:
is this condition needed? - `window.LC_API.on_chat_ended`

Reply:
Is optional chaining supported here?

Reply:
yeah. Theres a console error complaining because on_chat_ended doesn't exist in LC_API object. i dont believe on_chat_ended exists in the object until live chat is started then closed.

Reply:
but we are checking for this function ` window.LC_API.on_chat_ended` before defining it at the below line

Reply:
ah u r right i'll fix.